commands/service_execute.go

```diff
@@ -141,15 +141,20 @@ func (c *serviceExecuteCmd) getData(taskKey string, s *coreapi.Service, dataStru
 		return string(b), err
 	}
 
-	if c.jsonFile == "" {
-		if survey.AskOne(&survey.Input{Message: "Enter the filepath to the inputs"}, &c.jsonFile, nil) != nil {
-			return "", errors.New("no filepath given")
+	// see if task has no inputs.
+	for _, task := range s.Tasks {
```

Comment:
I remember that we had an argument about the similar thing that resulted with the discussion whether we should adopt client side validations for _commands_ pkg or not. I don't remember the result, do we have an agreement on that one? If no, let's talk about it on forum and apply what's decided.

Reply:
We already have a client-side validation in some parts and in general, a good practice is to validate input on client and server side if possible (unless you have really strong arguments why not to).

For example, we have json validation in getData.

Reply:
I see. /cc @antho1404 @NicolasMahe what do you think about this?

To me, I'd like to get errors directly from the Core to avoid maintaining same logic both in client and server side. To me, it's just an extra API call which is easy and fast to do. But with the _Network_ feature we may want to lower the pressure on Core by doing these validations on client side as well tho I'm still not sure it's that necessary.

But either way, I think core should handle validations nicely and provide errors in a common, detailed format so clients can show beautiful errors to users. E.g.:

calling _DeployService()_ with invalid mesg.yml:
```
{
  "error": {
    "message": "invalid mesg.yml",
    "errors": [
       { "field": "tasks.call.inputs.url.type", value: "123", message: "can be String, Object etc..." }
       ...
     ]
  }
}
```

A pattern similar to above _(error: (message), !optional!(errors: [field, value, message]) )_ can be applied to all gRRC APIs.

Reply:
I agree with @krhubert on this one. We could provide task validation as well. The command already load the service, it could also check if the task exist.
The function `taskKeysFromService` in the command return the all the tasks keys.

Reply:
@ilgooz please add the task key validation and then we merge

Reply:
done